frontend/src/component/map.vue

```diff
@@ -96,33 +128,90 @@ export default {
         this.map.on("load", () => {
           this.loaded = true;
           this.updatePosition();
+          this.map.resize();
         });
+
+        // Add click handler for interactive mode
+        if (this.clickable) {
+          this.map.on("click", (e) => {
+            const lat = e.lngLat.lat;
+            const lng = e.lngLat.lng;
+            this.$emit("map-clicked", { lat, lng });
+            this.$emit("update:latlng", [lat, lng]);
+          });
+        }
       } catch (error) {
         console.error("map: initialization failed", error);
         this.loaded = false;
       }
     },
     updatePosition() {
-      if (this.map && this.loaded) {
-        if (this.position[0] === this.lng && this.position[1] === this.lat) {
-          return;
+      if (!this.map || !this.loaded) {
+        return;
+      }
+
+      if (this.position[0] === this.latlng[1] && this.position[1] === this.latlng[0] && this.marker) {
+        return;
+      }
+
+      // Skip invalid or empty coordinates
+      if (!(this.latlng[0] && this.latlng[1] && !(this.latlng[0] === 0 && this.latlng[1] === 0))) {
+        if (this.marker) {
+          this.marker.remove();
+          this.marker = null;
         }
+        return;
+      }
 
-        this.position = [this.lng, this.lat];
+      this.position = [this.latlng[1], this.latlng[0]]; // Convert [lat, lng] to [lng, lat] for MapLibre
+
+      // Always center map when position changes in interactive mode
+      if (this.interactive) {
+        this.map.setCenter(this.position, {
+          zoom: this.zoom,
+          animate: false,
+        });
+      } else {
         this.map.setCenter(this.position);
+      }
 
-        if (this.marker) {
-          this.marker.setLngLat(this.position);
-        } else {
-          this.marker = new maplibregl.Marker({
-            color: "#3fb4df",
-            draggable: false,
-          })
-            .setLngLat(this.position)
-            .addTo(this.map);
+      if (this.marker) {
+        this.marker.setLngLat(this.position);
+      } else {
+        this.marker = new maplibregl.Marker({
+          color: "#3fb4df",
+          draggable: this.draggable,
+        })
+          .setLngLat(this.position)
+          .addTo(this.map);
+
+        // Add drag event listener for draggable markers
+        if (this.draggable) {
+          this.marker.on("dragend", () => {
+            const lngLat = this.marker.getLngLat();
+            this.$emit("marker-moved", { lat: lngLat.lat, lng: lngLat.lng });
+            this.$emit("update:latlng", [lngLat.lat, lngLat.lng]);
+          });
         }
       }
     },
+    // Public method to remove marker
+    removeMarker() {
+      if (this.marker) {
+        this.marker.remove();
+        this.marker = null;
+      }
+    },
+    // Public method to fly to coordinates
+    flyTo(lat, lng, zoom = this.zoom) {
+      if (this.map) {
+        this.map.flyTo({
```

Comment:
Does this restore the fly animation, when the coordinates change? I replaced it with a simple position change in the version released on Monday, as I couldn't find a way to speed up the animation in the time available.

Could you please check which map API parameters/options can be used to render the fly animation faster, especially for locations that are very far away? When map animations are disabled under Settings > General, no animation should be used at all. Thank you!